src/coreclr/vm/interoputil.cpp

```diff
@@ -1209,9 +1209,6 @@ void CleanupSyncBlockComData(InteropSyncBlockInfo* pInteropInfo)
     }
     CONTRACTL_END;
 
-    if ((g_fEEShutDown & ShutDown_SyncBlock) && IsAtProcessExit() )
```

Comment:
`ShutDown_SyncBlock` was never set in release build, so this condition was always false in the shipping bits. Better to align non-release builds to do the same